File: sortStringMyWay.cpp
// 프로그래머스 문자열 내 마음대로 정렬하기
#include <string>
#include <vector>
#include <algorithm>

using namespace std;
int globalN;
bool cmp (string a, string b){
    if (a[globalN] == b[globalN]){
        return a < b;
    }
    return a[globalN] < b[globalN];
}
vector<string> solution(vector<string> strings, int n) {
    globalN = n;
    vector<string> answer;
    sort(strings.begin(), strings.end(), cmp);
    answer = strings;
    return answer;
}

File: sortStringMyWay.js
// 프로그래머스 문자열 내 마음대로 정
function solution(strings, n) {
    var answer = strings.sort(function (a, b) {
        if (a[n] === b[n]) {
            return (a > b)?1:-1
        }
        else if (a[n] > b[n]) return 1;
        else return -1;
    })
    return answer;
}
